### PR TITLE
[API][Web][Forms][Workflow][SIMS-#1058]Tuition remittance

### PR DESCRIPTION
**Pull request by guru-aot**
**Acceptance criteria**
- [x] Remove "Tuition Remittance" from the offering (Form and DB)
- Enable Institution Users to request tuition remittance at the time of confirming enrolment, not to exceed the lesser of EITHER (Actual tuition + Program related costs) OR (Canada grants+Canada Loan+BC Loan)
- [x] Add remittance_amount column to disbursement schedule table
- [x] Add value to "confirm enrollment" UI to allow number to be entered
- [x] On confirm of enrollment, if the validation of that number fails and display toast message "Invalid tuition remittance amount"
- [x] On confirm of enrollment, if the validation of that number passes save the value to remittance_amount column
- [x] On disbursement of FT ecert file ensure remittance_amount is inserted into School Amount column
- [x] On disbursement of FT ecert file ensure Student Amount column is the difference between disbursement amount and school amount
- [x] On disbursement of PT ecert file ensure remittance_amount is inserted into School Amount column which inturn inserted into the EI remittance field

image

### CODE REVIEW

File: sources/packages/api/src-sql/DisbursementSchedules/Add-tuition-remittance.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+-- Add tuition_remittance_requested_amount
```
Comment (ann-aot) [inline sources/packages/api/src-sql/DisbursementSchedules/Add-tuition-remittance.sql:1]: full stop

File: sources/packages/api/src-sql/EducationProgramsOfferings/Add-tuition-remittance.sql [hunk 2]
```diff
@@ -0,0 +1,15 @@
+-- Add tuition_remittance_requested_amount
```
Comment (ann-aot) [inline sources/packages/api/src-sql/EducationProgramsOfferings/Add-tuition-remittance.sql:1]: full stop

File: sources/packages/api/src-sql/DisbursementSchedules/Add-tuition-remittance.sql [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+ALTER TABLE
+    sims.disbursement_schedules
+ADD
+    COLUMN IF NOT EXISTS tuition_remittance_requested_amount INT NOT NULL DEFAULT 0;
+
+COMMENT ON COLUMN sims.disbursement_schedules.tuition_remittance_requested_amount IS 'Tuition remittance Amount Requested';
```
Comment (andrewsignori-aot) [inline sources/packages/api/src-sql/DisbursementSchedules/Add-tuition-remittance.sql:7]: No need to "Amount Requested" to have capital letters and please add the period to the sentence.

File: sources/packages/api/src-sql/EducationProgramsOfferings/Add-tuition-remittance.sql (hunk 2 above)
Comment (andrewsignori-aot) [inline sources/packages/api/src-sql/EducationProgramsOfferings/Add-tuition-remittance.sql:1]: Please would be good to start the file name with "Rollback" to be easy to follow.

File: sources/packages/api/src/database/entities/disbursement-schedule.model.ts [hunk 4]
```diff
@@ -137,4 +137,12 @@ export class DisbursementSchedule extends RecordDataModel {
     referencedColumnName: ColumnNames.ID,
   })
   studentAssessment: StudentAssessment;
+  /**
+   * Tuition Remittance Amount Requested
```
Comment (ann-aot) [inline sources/packages/api/src/database/entities/disbursement-schedule.model.ts:141]: full stop

Comment (andrewsignori-aot) [inline sources/packages/api/src/database/entities/disbursement-schedule.model.ts:141]: Please convert it to an English sentence.
When possible it is also nice to add business context, for instance:
```
The tuition remittance amount is the value that is subtracted from the student grant/loan to be paid directly to the institution.
```

File: sources/packages/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.ts [hunk 5]
```diff
@@ -290,6 +293,7 @@ export class ConfirmationOfEnrollmentController extends BaseController {
   async confirmEnrollment(
     @Param("locationId") locationId: number,
     @Param("disbursementScheduleId") disbursementScheduleId: number,
+    @Body() payload: ConfirmationOfEnrollmentDto,
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.ts:296]: please follow the DTO naming format

File: sources/packages/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.ts [hunk 6]
```diff
@@ -329,11 +342,29 @@ export class ConfirmationOfEnrollmentController extends BaseController {
...
+    (Actual tuition + Program related costs) OR (Canada grants + Canada Loan + BC Loan). */
+    const offering = disbursementSchedule.studentAssessment.offering;
+    const offeringAmount =
+      offering.actualTuitionCosts + offering.programRelatedCosts;
+    const maxTuitionAllowed = Math.min(offeringAmount, disbursementAmount);
+
+    if (payload.tuitionRemittanceAmount > maxTuitionAllowed) {
+      throw new UnprocessableEntityException(
+        new ApiProcessError(
+          INVALID_TUITION_REMITTANCE_AMOUNT_MESSAGE,
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.ts:356]: pls add the errors in the api tag

File: sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-full-time-integration/e-cert-files/e-cert-file-record.ts [hunk 7]
```diff
@@ -50,7 +50,7 @@ export class ECertFullTimeFileRecord extends ECertFileRecord {
   disbursementAmount: number;
   /**
    * Dollar amount that should be sent to the student. This represents the total
-   * amount of BCSL/grants and CSL/grants that should go to the student on this document.
+   * disbursement amount minus school amount that should go to the student on this document.
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-full-time-integration/e-cert-files/e-cert-file-record.ts:53]: I would try to complete this as below
```
disbursement amount minus the school amount (a.k.a. tuition remittence) that should go to the student on this document.
```

File: sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-part-time-integration/e-cert-files/e-cert-file-record.ts [hunk 8]
```diff
@@ -190,7 +196,7 @@ export class ECertPartTimeFileRecord extends ECertFileRecord {
       record.appendWithEndFiller(this.emailAddress, 75, SPACE_FILLER);
       record.append("P"); // 'P' for part-time. Full time is done by another integration to another system.
       record.appendDate(this.enrollmentConfirmationDate, DATE_FORMAT);
-      record.repeatAppend(SPACE_FILLER, 7); // EI Remittance Amt, optional, not provided.
+      record.appendWithStartFiller(this.schoolAmount, 7, NUMBER_FILLER); // EI Remittance Amt, optional, not provided.
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-part-time-integration/e-cert-files/e-cert-file-record.ts:199]: The inline comment can be removed now.

File: sources/packages/api/src/services/disbursement-schedule-service/disbursement-schedule-service.ts [hunk 9]
```diff
@@ -515,8 +526,8 @@ export class DisbursementScheduleService extends RecordDataModelService<Disburse
       })
       .andWhere("application.applicationStatus IN (:...status)", {
         status: [ApplicationStatus.enrollment, ApplicationStatus.completed],
-      })
-      .getOne();
+      });
+    return query.getOne();
```
Comment (ann-aot) [inline sources/packages/api/src/services/disbursement-schedule-service/disbursement-schedule-service.ts:530]: mm, i think you can directly return line 497

File: sources/packages/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.ts [hunk 10]
```diff
@@ -61,6 +64,10 @@ const COE_NOT_FOUND_MESSAGE =
 const FIRST_COE_NOT_COMPLETE = "FIRST_COE_NOT_COMPLETE";
 const FIRST_COE_NOT_COMPLETE_MESSAGE =
   "First disbursement(COE) not complete. Please complete the first disbursement.";
+export const INVALID_TUITION_REMITTANCE_AMOUNT =
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.ts:67]: Please move to the error constants file.

File: sources/packages/forms/src/form-definitions/confirmsstudentenrollment.json [hunk 11]
```diff
@@ -1,4101 +1,3458 @@
...
+          "addons": []
+        },
+        {
+          "label": "Application Number | Location Name | Study Dates",
+          "tag": "p",
+          "className": "",
+          "attrs": [{ "attr": "", "value": "" }],
+          "content": "{{ data.applicationNumber + \" | \" + data.applicationLocationName + \" | \" + data.applicationOfferingStartDate + \" - \" + data.applicationOfferingEndDate }}",
+          "refreshOnChange": true,
+          "customClass": "category-header-medium secondary-color  mt-5 mb-5",
```
Comment (ann-aot) [inline sources/packages/forms/src/form-definitions/confirmsstudentenrollment.json:123]: there is a extra space before `mt-5`

File: sources/packages/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.ts (hunk 5 above)
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.ts:296]: Please name it accordingly.

File: sources/packages/forms/src/form-definitions/confirmsstudentenrollment.json [hunk 12]
```diff
@@ -1,4101 +1,3458 @@
...
+                  "addons": []
+                },
+                {
+                  "label": "programName",
+                  "tag": "p",
+                  "className": "",
+                  "attrs": [{ "attr": "", "value": "" }],
+                  "content": "{{ data.applicationProgramName }}",
+                  "refreshOnChange": true,
+                  "customClass": "font-weight-bold color-blue h4",
```
Comment (ann-aot) [inline sources/packages/forms/src/form-definitions/confirmsstudentenrollment.json:425]: is h4? then why `font-weight-bold`? is this style correct?

File: sources/packages/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.ts [hunk 13]
```diff
@@ -299,6 +303,15 @@ export class ConfirmationOfEnrollmentController extends BaseController {
         disbursementScheduleId,
       );
 
+    const disbursementAmount = getTotalDisbursementAmount(
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.ts:306]: Please move this logic closed to where it is used.

File: sources/packages/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.ts [hunk 14]
```diff
@@ -329,11 +342,29 @@ export class ConfirmationOfEnrollmentController extends BaseController {
       );
     }
 
+    /* Enable Institution Users to request tuition remittance at the time
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.ts:345]: Please keep the comment format aligned with the solution.

File: sources/packages/forms/src/form-definitions/confirmsstudentenrollment.json [hunk 15]
```diff
@@ -1,4101 +1,3458 @@
...
-                                    "placeholder": "",
-                                    "prefix": "",
-                                    "customClass": "",
-                                    "suffix": "",
-                                    "multiple": false,
-                                    "defaultValue": null,
-                                    "protected": false,
-                                    "unique": false,
-                                    "persistent": false,
-                                    "hidden":
```
Comment (ann-aot) [inline sources/packages/forms/src/form-definitions/confirmsstudentenrollment.json:1700]: `breakEndDate1`? can we remove `1`

Comment (ann-aot) [inline sources/packages/forms/src/form-definitions/confirmsstudentenrollment.json:2103]: pls remove `1`

Comment (ann-aot) [inline sources/packages/forms/src/form-definitions/confirmsstudentenrollment.json:2265]: pls remove `1`

Comment (ann-aot) [inline sources/packages/forms/src/form-definitions/confirmsstudentenrollment.json:2427]: please remove `1` from here and other fields too

File: sources/packages/api/src/route-controllers/confirmation-of-enrollment/models/confirmation-of-enrollment.model.ts [hunk 16]
```diff
@@ -37,14 +35,17 @@ export interface COEDeniedReasonDto {
...
 export class DenyConfirmationOfEnrollmentDto {
-  @IsNotEmpty()
-  @IsInt()
   @Min(1)
   coeDenyReasonId: number;
   @IsOptional()
   otherReasonDesc?: string;
 }
 
+export class ConfirmationOfEnrollmentDto {
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/confirmation-of-enrollment/models/confirmation-of-enrollment.model.ts:44]: Please adjust the name.

File: sources/packages/api/src/services/disbursement-schedule-service/disbursement-schedule-service.ts [hunk 17]
```diff
@@ -297,12 +297,14 @@ export class DisbursementScheduleService extends RecordDataModelService<Disburse
    * @param userId
    * @param applicationId
    * @param applicationStatus
+   * @param tuitionRemittanceRequestedAmount
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/services/disbursement-schedule-service/disbursement-schedule-service.ts:300]: Please add a comment.

File: sources/packages/web/src/services/http/ConfirmationOfEnrollmentApi.ts [hunk 18]
```diff
@@ -60,11 +61,16 @@ export class ConfirmationOfEnrollmentApi extends HttpBaseClient {
   public async confirmCOE(
     locationId: number,
     disbursementScheduleId: number,
+    confirmationData: ConfirmationOfEnrollment,
```
Comment (ann-aot) [inline sources/packages/web/src/services/http/ConfirmationOfEnrollmentApi.ts:64]: please follow the naming convention

File: sources/packages/web/src/views/institution/locations/confirmation-of-enrollment/ApplicationDetailsForCOE.vue [hunk 19]
```diff
@@ -90,20 +102,55 @@ export default {
...
+          payload,
+        );
+        toast.success("Confirmed", "Confirmation of Enrollment Confirmed!");
+      } catch (error: unknown) {
+        if (error instanceof ApiProcessError) {
+          if (error.errorType === FIRST_COE_NOT_COMPLETE) {
+            toast.error("First COE is not completed", error.message);
+            return;
+          }
+          if (error.errorType === INVALID_TUITION_REMITTANCE_AMOUNT) {
```
Comment (ann-aot) [inline sources/packages/web/src/views/institution/locations/confirmation-of-enrollment/ApplicationDetailsForCOE.vue:142]: you can use if-else or switch and avoid return

File: sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-part-time-integration/e-cert-files/e-cert-file-record.ts [hunk 20]
```diff
@@ -31,6 +31,12 @@ export class ECertPartTimeFileRecord extends ECertFileRecord {
    * ! Must be rounded, only integer, no decimals.
    */
   disbursementAmount: number;
+  /**
+   * Dollar amount that should be sent to the student. This represents the total
+   * school amount - tuition remittance amount that should go to the student on this document.
+   * ! Must be rounded, only integer, no decimals.
```
Comment (ann-aot) [inline sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-part-time-integration/e-cert-files/e-cert-file-record.ts:37]: you can use `* *` , IMO

File: sources/packages/web/src/services/http/ConfirmationOfEnrollmentApi.ts (hunk 18 above)
Comment (andrewsignori-aot) [inline sources/packages/web/src/services/http/ConfirmationOfEnrollmentApi.ts:64]: This is probably supposed to be named `APIInDTO`, right?

File: sources/packages/web/src/types/contracts/institution/ConfirmationOfEnrollment.ts [hunk 21]
```diff
@@ -68,6 +66,10 @@ export interface DenyConfirmationOfEnrollment {
   otherReasonDesc?: string;
 }
 
+export interface ConfirmationOfEnrollment {
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/types/contracts/institution/ConfirmationOfEnrollment.ts:69]: Is this the DTO? Can we please rename it and move it to the proper folder?

File: sources/packages/web/src/views/institution/locations/confirmation-of-enrollment/ApplicationDetailsForCOE.vue [hunk 22]
```diff
@@ -39,19 +45,25 @@
...
+  ApiProcessError,
+  ConfirmationOfEnrollment,
 } from "@/types";
-import ConfirmCOE from "@/components/institutions/confirmation-of-enrollment/modals/ConfirmCOEModal.vue";
 import ConfirmCOEEditModal from "@/components/institutions/confirmation-of-enrollment/modals/ConfirmCOEEditModal.vue";
 import ConfirmCOEDenyModal from "@/components/institutions/confirmation-of-enrollment/modals/ConfirmCOEDenyModal.vue";
-import { useToastMessage, ModalDialog } from "@/composables";
+import { useToastMessage, ModalDialog, useFormioUtils } from "@/composables";
 import Information from "@/components/institutions/confirmation-of-enrollment/information.vue";
+export const FIRST_COE_NOT_COMPLETE = "FIRST_COE_NOT_COMPLETE";
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/institution/locations/confirmation-of-enrollment/ApplicationDetailsForCOE.vue:64]: Please move to the correct error-code constant files.

File: sources/packages/web/src/views/institution/locations/confirmation-of-enrollment/ApplicationDetailsForCOE.vue [hunk 23]
```diff
@@ -39,19 +45,25 @@
...
+export const INVALID_TUITION_REMITTANCE_AMOUNT =
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/institution/locations/confirmation-of-enrollment/ApplicationDetailsForCOE.vue:65]: Please move to the correct error-code constant files.

File: sources/packages/web/src/views/institution/locations/confirmation-of-enrollment/ApplicationDetailsForCOE.vue [hunk 24]
```diff
@@ -90,20 +102,55 @@ export default {
...
         );
     };
-
+    const showHideConfirmCOE = async () => {
+      const modalResult = await confirmCOEModal.value.showModal();
+      if (!modalResult) {
+        return;
+      }
+      try {
+        const tuitionRemittanceAmount = formioUtils.getComponentValueByKey(
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/institution/locations/confirmation-of-enrollment/ApplicationDetailsForCOE.vue:123]: The `modalResult` has the form.io submitted data, please get the value from there and remove the `TUITION_REMITTANCE_AMOUNT`.

File: sources/packages/web/src/views/institution/locations/confirmation-of-enrollment/ApplicationDetailsForCOE.vue [hunk 25]
```diff
@@ -90,20 +102,55 @@ export default {
...
+          tuitionRemittanceAmount,
+        };
+        await ConfirmationOfEnrollmentService.shared.confirmCOE(
+          props.locationId,
+          props.disbursementScheduleId,
+          payload,
+        );
+        toast.success("Confirmed", "Confirmation of Enrollment Confirmed!");
+      } catch (error: unknown) {
+        if (error instanceof ApiProcessError) {
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/institution/locations/confirmation-of-enrollment/ApplicationDetailsForCOE.vue:137]: I believe that a `switch` would look better here, but it is just a suggestion.

File: sources/packages/web/src/views/institution/locations/confirmation-of-enrollment/ApplicationDetailsForCOE.vue [hunk 26]
```diff
@@ -90,20 +102,55 @@ export default {
...
+          if (error.errorType === FIRST_COE_NOT_COMPLETE) {
+            toast.error("First COE is not completed", error.message);
+            return;
+          }
+          if (error.errorType === INVALID_TUITION_REMITTANCE_AMOUNT) {
+            toast.error("Invalid tuition remittance amount", error.message);
+            return;
+          }
+        } else {
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/institution/locations/confirmation-of-enrollment/ApplicationDetailsForCOE.vue:146]: In the current implementation the `else` can be removed, right?

**Review by andrewsignori-aot — COMMENTED** [review]
Great work, please take a look at the comments.

File: sources/packages/api/src/database/entities/disbursement-schedule.model.ts (hunk 4 above)
Comment (dheepak-aot) [reply sources/packages/api/src/database/entities/disbursement-schedule.model.ts:141]: Just checking, in that case should the comment in db column should also be updated?

File: sources/packages/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.ts [hunk 27]
```diff
@@ -61,6 +64,10 @@ const COE_NOT_FOUND_MESSAGE =
 const FIRST_COE_NOT_COMPLETE = "FIRST_COE_NOT_COMPLETE";
 const FIRST_COE_NOT_COMPLETE_MESSAGE =
   "First disbursement(COE) not complete. Please complete the first disbursement.";
+export const INVALID_TUITION_REMITTANCE_AMOUNT =
+  "INVALID_TUITION_REMITTANCE_AMOUNT";
+const INVALID_TUITION_REMITTANCE_AMOUNT_MESSAGE =
+  "Tuition amount provided should be lesser of EITHER (Actual tuition + Program related costs) OR (Canada grants + Canada Loan + BC Loan).";
```
Comment (dheepak-aot) [inline sources/packages/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.ts:70]: Just checking on the given message against business logic. Indeed the tuition remittance  must be lesser then both (Actual tuition + Program related costs) AND (Canada grants + Canada Loan + BC Loan). 

We pick the min value between  (Actual tuition + Program related costs) and (Canada grants + Canada Loan + BC Loan). So if lesser than one it implicitly is lesser than both.

File: sources/packages/web/src/views/institution/locations/confirmation-of-enrollment/ApplicationDetailsForCOE.vue [hunk 28]
```diff
@@ -90,20 +102,55 @@ export default {
...
+      if (!modalResult) {
+        return;
+      }
+      try {
+        const tuitionRemittanceAmount = formioUtils.getComponentValueByKey(
+          modalResult,
+          TUITION_REMITTANCE_AMOUNT,
+        );
+        const payload: ConfirmationOfEnrollment = {
+          tuitionRemittanceAmount,
```
Comment (dheepak-aot) [inline sources/packages/web/src/views/institution/locations/confirmation-of-enrollment/ApplicationDetailsForCOE.vue:128]: According to form, the tuitionRemittanceAmount is available in the payload only when YES is selected to request tuition remittance. What happens when NO is selected. 

tuitionRemittanceAmount will be undefined right?  (also this field(tuitionRemittanceAmount) is mandatory at Client and API DTO)

**Comment by dheepak-aot** [comment]
Nice work @guru-aot 👍. Added few comments.

File: sources/packages/forms/src/form-definitions/confirmcoe.json [hunk 29]
```diff
@@ -0,0 +1,611 @@
...
+              "attributes": {},
+              "overlay": {
+                "style": "",
+                "page": "",
+                "left": "",
+                "top": "",
+                "width": "",
+                "height": ""
+              },
+              "type": "number",
```
Comment (dheepak-aot) [inline sources/packages/forms/src/form-definitions/confirmcoe.json:387]: Are we considering to add data-cy?

File: sources/packages/forms/src/form-definitions/confirmsstudentenrollment.json (hunk 11 above)
Comment (guru-aot) [reply sources/packages/forms/src/form-definitions/confirmsstudentenrollment.json:123]: Not part of my Changes.

File: sources/packages/forms/src/form-definitions/confirmsstudentenrollment.json (hunk 12 above)
Comment (guru-aot) [reply sources/packages/forms/src/form-definitions/confirmsstudentenrollment.json:425]: Not part of my Changes.

File: sources/packages/forms/src/form-definitions/confirmsstudentenrollment.json (hunk 15 above)
Comment (guru-aot) [reply sources/packages/forms/src/form-definitions/confirmsstudentenrollment.json:1700]: Not part of my Changes.

Comment (guru-aot) [reply sources/packages/forms/src/form-definitions/confirmsstudentenrollment.json:2103]: Not part of my Changes.

Comment (guru-aot) [reply sources/packages/forms/src/form-definitions/confirmsstudentenrollment.json:2265]: Not part of my Changes.

Comment (guru-aot) [reply sources/packages/forms/src/form-definitions/confirmsstudentenrollment.json:2427]: Not part of my Changes.

File: sources/packages/web/src/views/institution/locations/confirmation-of-enrollment/ApplicationDetailsForCOE.vue (hunk 28 above)
Comment (guru-aot) [reply sources/packages/web/src/views/institution/locations/confirmation-of-enrollment/ApplicationDetailsForCOE.vue:128]: default is 0

File: sources/packages/api/src/database/entities/disbursement-schedule.model.ts (hunk 4 above)
Comment (andrewsignori-aot) [reply sources/packages/api/src/database/entities/disbursement-schedule.model.ts]: > Just checking, in that case should the comment in db column should also be updated?

I would prefer to keep both comments always synched unless there is a reason to not do it. For this case the DB comment is not bad, so it is not a blocker IMO.

File: sources/packages/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.ts [hunk 30]
```diff
@@ -283,13 +289,18 @@ export class ConfirmationOfEnrollmentController extends BaseController {
    * @param locationId location id of the application
    * @param disbursementScheduleId disbursement schedule id of COE
    */
+  @ApiUnprocessableEntityResponse({
+    description:
+      "Tuition amount provided should be lesser than both (Actual tuition + Program related costs) and (Canada grants + Canada Loan + BC Loan). OR First disbursement(COE) not complete. Please complete the first disbursement.",
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.ts:294]: Please convert to a regular sentence (`Loan + BC Loan). OR First disbursement(COE)`???).

Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.ts:294]: The const `INVALID_TUITION_REMITTANCE_AMOUNT_MESSAGE` can also be used here.

File: sources/packages/web/src/services/ConfirmationOfEnrollmentService.ts [hunk 31]
```diff
@@ -8,6 +8,7 @@ import {
   PaginationOptions,
 } from "@/types";
 import ApiClient from "./http/ApiClient";
+import { ConfirmationOfEnrollmentAPIInDTO } from "./http/dto/ConfirmationOfEnrolment.dto";
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/services/ConfirmationOfEnrollmentService.ts:11]: Please use "@".

File: sources/packages/web/src/services/http/ConfirmationOfEnrollmentApi.ts [hunk 32]
```diff
@@ -10,6 +10,7 @@ import {
 } from "@/types";
 import HttpBaseClient from "./common/HttpBaseClient";
 import { addSortOptions, addPaginationOptions } from "@/helpers";
+import { ConfirmationOfEnrollmentAPIInDTO } from "./dto/ConfirmationOfEnrolment.dto";
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/services/http/ConfirmationOfEnrollmentApi.ts:13]: Please use "@".

**Review by andrewsignori-aot — COMMENTED** [review]
Thanks for doing the changes. Minor comments only.

**Review by andrewsignori-aot — APPROVED** [review]
Thanks for doing the changes, looks good 👍

**Review by ann-aot — APPROVED** [review]
Good work @guru-aot  👍  Thanks for doing the changes

**Review by henryoforeh-aot — APPROVED** [review]

**Review by dheepak-aot — APPROVED** [review]
Thanks for doing the changes @guru-aot . LGTM.

**Pull request merged**